clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -1281,30 +1281,36 @@ private static byte checkRange(final byte i) {
 
     /**
      * Convert timestamp to an epoch value
-     * @param timestamp to be converted
-     * @return epoch value of a given timestamp
-     * @throws ParseException for timestamp that doesn't follow ISO8601 format
-     */
-    public static long getDateTime(String timestamp) throws ParseException {
+     * @param timestamp the timestamp to be converted, the accepted formats are: 2020-11-10T16:51:38.145, 2020-11-10T16:51:38.198+0800,
+     *                  2020-11-10T16:51:38.198+08, 2020-11-10T16:51:38.199+0800, and 2020-11-10T16:51:38.199+08:00
+     * @return epoch value of a given timestamp (i.e. the number of milliseconds since January 1, 1970, 00:00:00 GMT)
+     * @throws ParseException for timestamp that doesn't follow ISO8601 format or the format is not expected
+     */
+    public static long getDateTime(String timestamp) throws ParseException, IllegalArgumentException {
+        if (timestamp == null) {
+            throw new IllegalArgumentException("Error parsing timestamp with null value");
+        }
+
         final String[] timestampParts = timestamp.split("T");
         if (timestampParts.length < 2) {
             throw new ParseException("Error parsing timestamp. It does not contain a 'T' according to ISO8601 format", timestamp.length());
         }
 
         final String secondPart = timestampParts[1];
-        if (secondPart == null || secondPart.isEmpty()) {
-            throw new ParseException("Error parsing timestamp. Time part after 'T' is null or empty", timestamp.length());
```

Comment:
It'll never be null or empty for the `secondPart` because we split("T") without the 2nd argument, and it'll call with `split("T", 0)`. And from the javadoc:
> If n is zero then the pattern will be applied as many times as possible, the array can have any length, and trailing empty strings will be discarded. 

https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#split-java.lang.String-int-